src/library_syscall.js

```diff
@@ -544,7 +588,13 @@ var SyscallsLibrary = {
       var flags = SYSCALLS.DEFAULT_POLLMASK;
 
       if (stream.stream_ops.poll) {
-        flags = stream.stream_ops.poll(stream);
+        var timeoutInMillis = -1;
+        if (timeout) {
+          var timeoutLow = (readfds ? {{{ makeGetValue('timeout', 0, 'i32') }}} : 0),
+              timeoutHigh = (readfds ? {{{ makeGetValue('timeout', 4, 'i32') }}} : 0);
```

Comment:
I think you can use ` var timeout = {{{ makeGetValue('timeout', 4, 'i53') }}}` instead of reading the hi/low pair yourself.

Reply:
Oh sorry, this is intentional here. The code reads the following two fields of `struct timeval`.

```
struct timeval {
  long tv_sec;   // seconds
  long tv_usec; // microseconds
};
```

The names `timeoutLow` and `timeoutHigh` were bad. I'll change them as `timeoutSeconds` and `timeoutMicroseconds`.

Reply:
I changed the variable names d8efb50a537df5d1179c6af79b1013a9b2a6185a

Reply:
In that case I wonder if its worth using `timeval.tv_sec` and `timeval.tv_usec` here rather than hardcoding 0 and 4?  Maybe not worth it?

Reply:
You are absolutely right. Why didn't I notice it? Done at 8fe4bb834a4936ce2f73bd1ad692eea95ddad070